modules/api/src/main/java/org/apache/fluo/api/config/FluoConfiguration.java

```diff
@@ -413,27 +438,46 @@ public int getConnectionRetryTimeout() {
     return retval;
   }
 
+  /**
+   * Sets the Apache Accumulo instance property {@value #ACCUMULO_INSTANCE_PROP} 
+   * @param accumuloInstance The instance to connect to, must not be empty
+   */
   public FluoConfiguration setAccumuloInstance(String accumuloInstance) {
     return setNonEmptyString(ACCUMULO_INSTANCE_PROP, accumuloInstance);
   }
-
+  /**
+   * Gets the Apache Accumulo instance
+   */
   public String getAccumuloInstance() {
     return getDepNonEmptyString(ACCUMULO_INSTANCE_PROP, CLIENT_ACCUMULO_INSTANCE_PROP);
   }
-
+  /**
+   * Sets the value of the property {@value #ACCUMULO_USER_PROP}
+   * @param accumuloUser The user name to use, must not be null.
+   */
   public FluoConfiguration setAccumuloUser(String accumuloUser) {
     return setNonEmptyString(ACCUMULO_USER_PROP, accumuloUser);
   }
 
+  /**
+   * Gets the value of the property {@value #ACCUMULO_USER_PROP}
+   */
   public String getAccumuloUser() {
     return getDepNonEmptyString(ACCUMULO_USER_PROP, CLIENT_ACCUMULO_USER_PROP);
   }
-
+  /**
+   * Sets the Apache Accumulo password property {@value #ACCUMULO_PASSWORD_PROP}
+   * @param accumuloPassword The password to use, must not be null.
+   */
   public FluoConfiguration setAccumuloPassword(String accumuloPassword) {
     setProperty(ACCUMULO_PASSWORD_PROP, verifyNotNull(ACCUMULO_PASSWORD_PROP, accumuloPassword));
     return this;
   }
 
+  /**
+   * Gets the Apache Accumulo password property value {@value #ACCUMULO_PASSWORD_PROP}
+   * @throws NoSuchElementException if {@value #ACCUMULO_PASSWORD_PROP} is not set
```

Comment:
very nice addition! knowing what runtime exceptions may thrown is so important for javadoc